scripts/generate/templates/Checkpoints.js

```diff
@@ -45,18 +53,27 @@ function push(
     return _insert(self.${opts.checkpointFieldName}, key, value);
 }
 
+/**
+ * @dev Return the value most recent checkpoint which key is lower or equal than the search key.
+ */
 function lowerLookup(${opts.historyTypeName} storage self, ${opts.keyTypeName} key) internal view returns (${opts.valueTypeName}) {
     uint256 length = self.${opts.checkpointFieldName}.length;
     uint256 pos = _lowerBinaryLookup(self.${opts.checkpointFieldName}, key, 0, length);
     return pos == length ? 0 : _unsafeAccess(self.${opts.checkpointFieldName}, pos).${opts.valueFieldName};
 }
 
+/**
+ * @dev Return the value oldest checkpoint which key is higher or equal than the search key.
```

Comment:
I think you got the docs for upper and lower reversed.

```suggestion
 * @dev Returns the value in the most recent checkpoint with key lower or equal than the search key.
```